app/models/utils/nav/customization_parser.rb

```diff
@@ -5,10 +5,45 @@ module Nav
     # Parses the top menu customization setting
     class CustomizationParser
       attr_accessor :markdown
+      attr_accessor :customizations
 
       def initialize(markdown)
         # a nil input is stored as an empty string
         self.markdown = markdown || ""
+        self.customizations = parse
+        @done = []
+      end
+
+      def filter_item(name:, path:, permitted:, icon:)
+        # check to see if this item has been customized
+        result = {}
+        result[:name] = name
+        translated_name = I18n.t("nav_links.main.#{name}")
+        if @customizations[translated_name]
+          result[:path] = @customizations[translated_name]
+          @done << translated_name
+        else
+          result[:path] = path
+        end
+        result[:permitted] = permitted
+        result[:icon] = icon
+        if result[:path] == "none"
+          nil
+        else
+          result
+        end
+      end
+
+      def each
+        @customizations.each do |c|
+          next if @done.include?(c[0])
+          result = {}
+          result[:name] = c[0]
+          result[:path] = c[1]
+          result[:permitted] = true
+          result[:icon] = "info-circle"
+          yield(result)
+        end
```

Comment:
Nice use of yield here! But I think you don't need the iterator pattern here. You could just just have a method called `extra_items` that maps the still-remaining `customizations` to item-style hashes and returns them, or `[]` if none remain. Then in the builder class you can just do `items.concat(customizer.extra_items)`.

Reply:
it's done.